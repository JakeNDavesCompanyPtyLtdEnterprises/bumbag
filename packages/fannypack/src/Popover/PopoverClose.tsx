import * as React from 'react';
import * as PropTypes from 'prop-types';
import { ButtonProps } from '../Button/Button';

import { PopoverClose as _PopoverClose } from './styled';
import PopoverHide from './PopoverHide';
import Icon from '../Icon';
import VisuallyHidden from '../VisuallyHidden';
import { Omit } from '../types';

export type LocalPopoverCloseProps = {
  hide(): void;
};
export type PopoverCloseProps = Omit<ButtonProps, 'children'> & LocalPopoverCloseProps;

export const PopoverClose: React.FunctionComponent<LocalPopoverCloseProps> = props => (
  <_PopoverClose use={PopoverHide} size="small" kind="link" {...props}>
    <VisuallyHidden>Close</VisuallyHidden>
    <Icon a11yHidden icon="times" />
  </_PopoverClose>
);

PopoverClose.propTypes = {
  hide: PropTypes.func.isRequired
};
PopoverClose.defaultProps = {};

// @ts-ignore
const C: React.FunctionComponent<PopoverCloseProps> = PopoverClose;
export default C;
