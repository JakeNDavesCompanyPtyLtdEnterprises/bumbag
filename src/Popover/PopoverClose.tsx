import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { ButtonProps } from '../Button/Button';

import { PopoverClose as _PopoverClose } from './styled';
import PopoverHide from './PopoverHide';
import Icon from '../Icon';
import VisuallyHidden from '../VisuallyHidden';
import { Omit } from '../types';

export interface LocalPopoverCloseProps {
  hide(): void;
}
export type PopoverCloseProps = Omit<ButtonProps, 'children'> & LocalPopoverCloseProps;

export const PopoverClose: React.FunctionComponent<LocalPopoverCloseProps> = props => (
  <_PopoverClose use={PopoverHide} size="small" kind="link" {...props}>
    <VisuallyHidden>Close</VisuallyHidden>
    <Icon a11yLabel="Close" aria-hidden="true" icon="times" />
  </_PopoverClose>
);

PopoverClose.propTypes = {
  hide: PropTypes.func.isRequired
};
PopoverClose.defaultProps = {};

// @ts-ignore
const C: React.FunctionComponent<PopoverCloseProps> = PopoverClose;
export default C;
