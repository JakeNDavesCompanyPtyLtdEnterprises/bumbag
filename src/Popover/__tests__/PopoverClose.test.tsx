import * as React from 'react';
import render from '../../_utils/tests/render';
import PopoverClose from '../PopoverClose';

it('renders correctly', () => {
  const { container } = render(<PopoverClose hide={jest.fn()}>Close</PopoverClose>);
  expect(container.firstChild).toMatchSnapshot();
});
