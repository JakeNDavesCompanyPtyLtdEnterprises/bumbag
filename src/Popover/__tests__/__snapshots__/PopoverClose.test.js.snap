// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c1[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c1 {
  font-size: 0.8em;
  height: 2em;
  padding: 0 0.5rem;
}

.c1 {
  border: 0;
  background: unset;
  color: #435a6f;
  fill: #435a6f;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:hover {
  color: #000;
  fill: #000;
}

.c0 {
  position: absolute;
  width: 24px;
  height: 24px;
  right: 0;
  top: 0;
  padding: 0px;
  text-align: right;
}

.c5 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.c3 {
  border: 0px;
  -webkit-clip: rect(0px,0px,0px,0px);
  clip: rect(0px,0px,0px,0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  position: absolute;
}

<button
  aria-haspopup="true"
  class="c0 c1 c2 "
  kind="link"
  style="size: small;"
  type="button"
>
  <div
    class="c3 c4"
  >
    Close
  </div>
  <svg
    aria-hidden="true"
    class="c5 c4"
    role="img"
    size="1"
    style="size: 1px;"
    viewBox="0 0 16 16"
  >
    <path
      d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L9.41 8z"
      fill-rule="even-odd"
    />
  </svg>
</button>
`;
